src/main/java/uk/gov/hmcts/ccd/domain/service/getevents/AuthorisedGetEventsOperation.java

```diff
@@ -74,12 +85,19 @@ public AuthorisedGetEventsOperation(@Qualifier("classified") GetEventsOperation
             throw new ValidationException("Cannot find case type definition for  " + caseTypeId);
         }
 
-        Set<String> userRoles = userRepository.getUserRoles();
-        if (userRoles == null || userRoles.isEmpty()) {
+        if (accessRoles == null || accessRoles.isEmpty()) {
             throw new ValidationException("Cannot find user roles for the user");
         }
 
-        return verifyReadAccess(events, userRoles, caseType);
+        return verifyReadAccess(events, accessRoles, caseType);
+    }
+
+    private Set<String> getAccessRoles(String caseId) {
+        return Sets.union(userRepository.getUserRoles(),
+            caseUserRepository
+                .findCaseRoles(Long.valueOf(caseId), userRepository.getUserId())
```

Comment:
we should add this logic to a new method in `CaseAccessService` alongside the similar methods we have there already

Reply:
Ok... I am going to move this to CaseAccessService, I need to update test cases as well for this.. I am working on this